cmd/argocd/commands/cluster.go

```diff
@@ -185,6 +200,72 @@ func getRestConfig(pathOpts *clientcmd.PathOptions, ctxName string) (*rest.Confi
 	return conf, nil
 }
 
+// NewClusterEditCommand returns a new instance of an `argocd cluster edit` command
+func NewClusterEditCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var (
+		clusterOptions cmdutil.ClusterOptions
+		clusterName    string
+	)
+	var command = &cobra.Command{
+		Use:   "edit cluster name",
+		Short: "Edit cluster information",
+		Example: `  # Edit cluster information
+  argocd cluster edit cluster-name --name new-cluster-name --namespaces all
+  argocd cluster edit cluster-name --name new-cluster-name --namespaces namespace-one,namespace-two`,
+		Run: func(c *cobra.Command, args []string) {
+			ctx := c.Context()
+			if len(args) != 1 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			// name of the cluster whose fields have to be edited.
+			clusterName = args[0]
+			conn, clusterIf := headless.NewClientOrDie(clientOpts, c).NewClusterClientOrDie()
+			defer io.Close(conn)
+			// checks the fields that needs to be edited
+			updatedFields := checkFieldsToUpdate(clusterOptions)
+			namespaces := clusterOptions.Namespaces
+			// check if all namespaces have to be considered
+			if len(namespaces) == 1 && strings.EqualFold(namespaces[0], allNamespaces) {
+				namespaces[0] = ""
+			}
+			if updatedFields != nil {
```

Comment:
1. Updated fields can never be empty. Should check if this is empty or not.  
2. What if same data is re-entered, should we fire the update request. Should we have a local check of what is currently and what not ?